tests: update kernel objects tests

maksimmasalski opened the pull request:
1. Add code change to the test_permission_inheritance() to let it
test that child thread can't access parent thread object. Now that test
tests one more related to it feature.
2. Add new Doxygen tags with informative descriptions about the kernel
objects tests. That will make reading and understanding kernel object
tests code easier.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

maksimmasalski commented:
@nashif @andrewboie @andyross I'm looking for code review of that PR. There is no serious code changes.

LeiW000 reviewed (approved):
It looks good to me.

nashif commented:
rebase

Commit:
tests: update kernel objects tests

1. Add code change to the test_permission_inheritance() to let it
test that child thread can't access parent thread object. Now that test
tests one more related to it feature.
2. Add new Doxygen tags with informative descriptions about the kernel
objects tests. That will make reading and understanding kernel object
tests code easier.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

maksimmasalski commented:
> rebase

Done.
